crates/bevy_ecs/src/world/mod.rs

```diff
@@ -644,16 +644,21 @@ impl World {
         Some(unsafe { ptr.cast::<T>().read() })
     }
 
+    /// Returns `true` if a resource of a given [`TypeId`] exists. Otherwise returns `false`.
+    #[inline]
+    pub fn contains_resource_type_id(&self, type_id: TypeId) -> bool {
```

Comment:
I think this should take a `ComponentId` instead of a `TypeId` to support future dynamic resources.

I expect that most functions will have a rust-native version with generics that is easy to use, and a dynamic version taking ids which will be useful for scripting and editors, for which a `TypeId` may not work.

Reply:
While I agree with the long term vision, all reflection in bevy currently works on `TypeId`s only. It will need a dedicated PR to move it to `ComponentId`s (which I might tackle at some point).
The corresponding functions for components have two versions, so I could also add a `fn contains_resource_id(&self, id: ComponentId)` in this PR even though I won't be using it.

Reply:
My reasoning for exposing the `ComponentId` variant is that you can use it to lookup a type id (by doing `components.get_resource_id` yourself) but you can't lookup a `ComponentId` using the `TypeId` variant.

Reply:
Fair enough, added in 8503ab7